src/python_extension.ts

```diff
@@ -105,60 +112,115 @@ export class PythonExtension {
 		});
 	}
 
-	async updateInmantaEnvVisibility(documentURI?) {
+	/**
+	 * Updates the visibility and text of the inmantaEnvSelector based on the current Python environment and active editor.
+	 * @param {Uri} [documentURI] - The URI of the document to check the workspace folder for.
+	 */
+	async updateInmantaEnvVisibility(documentURI?: Uri): Promise<void> {
 		let venvName = this.virtualEnvName;
-		let folderName = "";
+		let folderName = this.getFolderName(documentURI);
+
+		if (documentURI) {
+			venvName = this.getVenvNameFromDocumentURI(documentURI) || venvName;
+		}
+
+		const version = await this.getPythonVersion();
+		const text = this.getSelectorText(version, venvName, folderName);
+
+		this.updateEnvSelector(text);
+	}
+
+	/**
+	 * Retrieves the folder name based on the document URI or the last active folder.
+	 * @param {Uri} [documentURI] - The URI of the document to check the workspace folder for.
+	 * @returns {string} - The folder name.
+	 */
+	private getFolderName(documentURI?: Uri): string {
 		const lastActiveFolder = getLastActiveFolder();
 		if (lastActiveFolder) {
-			folderName = lastActiveFolder.name;
+			return lastActiveFolder.name;
 		}
+
 		if (documentURI) {
-			try{
-				const folder = workspace.getWorkspaceFolder(documentURI);
-				if (folder) {
-					folderName = folder.name;
-					venvName = this.pythonPathToEnvName(getLanguageMap().get(folder.uri.toString()).pythonPath);
-				}
+			const folder = workspace.getWorkspaceFolder(documentURI);
+			if (folder) {
+				return folder.name;
 			}
+		}
+
+		return "";
+	}
 
-			catch (error){
-				console.error(`Failed to get Python version:` + error);
+	/**
+	 * Retrieves the virtual environment name from the document URI.
+	 * @param {Uri} documentURI - The URI of the document to check the workspace folder for.
+	 * @returns {string | undefined} - The virtual environment name or undefined if not found.
+	 */
+	private getVenvNameFromDocumentURI(documentURI: Uri): string | undefined {
+		try {
+			const folder = workspace.getWorkspaceFolder(documentURI);
+			if (folder) {
+				return this.pythonPathToEnvName(getLanguageMap().get(folder.uri.toString()).pythonPath);
 			}
-			
+		} catch (error) {
+			console.error(`Failed to get virtual environment name: ${error}`);
 		}
+		return undefined;
+	}
```

Comment:
Can this have a default value as an argument (this.virtualEnvName) so we can return that instead of undefined